src/zcbor_decode.c

```diff
@@ -437,9 +429,9 @@ bool zcbor_map_start_decode(zcbor_state_t *state)
 {
 	bool ret = list_map_start_decode(state, ZCBOR_MAJOR_TYPE_MAP);
 
-	if (ret && !INDET_LEN(state->elem_count)) {
-		if (INDET_LEN(state->elem_count * 2)) {
-			/* The new elem_count interferes with the INDET_LEN bit. */
+	if (ret && !state->indefinite_length_array) {
+		if (state->elem_count >= (MAX_ELEM_COUNT / 2)) {
+			/* The new elem_count is too large. */
```

Comment:
Why? Don't we have a bool to keep track of this so that we can have bigger values which now don't interfece with INDETL_LEN bit?